src/accel/cuda/GPUSolver.cu

```diff
@@ -397,24 +397,24 @@ __global__ void computeFSRScatterSourcesOnDevice(int* FSR_materials,
 
 
 /**
- * @brief Perform an atomic addition in double precision to an array address.
- * @details This method is straight out of CUDA C Developers Guide (cc 2013).
- * @param address the array memory address
- * @param val the value to add to the array
- * @return the atomically added array value and input value
+ * @brief Set all FSR spectra to match chi of a given material
+ * @param chi_material pointer to device material's chi spectrum to use
+ * @param scalar_flux the array of FSR scalar fluxes
  */
-__device__ double atomicAdd(double* address, double val) {
-
-  unsigned long long int* address_as_ull = (unsigned long long int*)address;
-  unsigned long long int old = *address_as_ull, assumed;
-
-  do {
-    assumed = old;
-    old = atomicCAS(address_as_ull, assumed, __double_as_longlong(val +
-                    __longlong_as_double(assumed)));
-  } while (assumed != old);
+__global__ void flattenFSRFluxesChiSpectrumOnDevice(dev_material* chi_material,
+                                                    FP_PRECISION* scalar_flux) {
+  /* NOTE, if this needs to go faster, copy Chi spectrum to each
+     thread block, then fill fluxes from there. Now, memory contention
+     may arise when many threads access elements of chi here at the
+     ~same~ time. */
 
-  return __longlong_as_double(old);
+  int tid = threadIdx.x + blockIdx.x * blockDim.x;
+  while (tid < *num_FSRs) {
```

Comment:
Are you sure here? num_FSRs is a length 1 array here so that it could be placed in GPU constant memory, which is why you access it like a pointer. Although, with this tiny example program, it seems that maybe in Will Boyd's day, __constant__ memory had to consist of arrays or pointers but now it can be values? I'm going to try changing these all to values rather than length one arrays.

```cuda
__constant__ unsigned asdf = 5;

__global__ void printSecretMessage()
{
  printf("Waddup %i\n", asdf);
}

int main()
{
  printSecretMessage<<<1,32>>>();
  cudaDeviceSynchronize();
}
```

Reply:
Well if it's an array you're right!
But let's make it a value.

Reply:
Your wish is my command! (done) I imagine this has to do with some old version of CUDA. I checked cudaMemcpyToSymbol with plain old values and it seems to work fine.